sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -42,6 +42,28 @@ options { tokenVocab = SqlBaseLexer; }
   public boolean double_quoted_identifiers = false;
 }
 
+compoundOrSingleStatement
+    : singleStatement
+    | singleCompound
+    ;
+
+singleCompound
```

Comment:
can you rename this to `singleCompoundStatement`

Reply:
So, if we want to be consistent with `statement` - `singleStatement` naming, then this should be named `singleBeginEndCompoundBlock`, but that seemed too much to me.
By this logic, `singleCompoundStatement` would mean it's a single `compoundStatement` which it is not.

Due to this two reasons I left only `singleCompound`. We can change it, but I would like to figure out better naming for other nodes, which I have no better idea than what's already proposed.